lld/ELF/Arch/RISCV.cpp

```diff
@@ -115,6 +115,24 @@ static uint32_t getEFlags(InputFile *f) {
   return cast<ObjFile<ELF32LE>>(f)->getObj().getHeader()->e_flags;
 }
 
+static bool hasExecutableSection(InputFile *f) {
+  if (config->is64) {
+    auto of = cast<ObjFile<ELF64LE>>(f);
+    auto sections = CHECK(of->getObj().sections(), of);
+    for (const auto section : sections)
+      if (section.sh_flags & SHF_EXECINSTR)
+        return true;
```

Comment:
Upstream would probably prefer braces round the outer for loop (and repeated below).

Reply:
I guess that depends on the reviewer, I've submitted patches with braces in the past and it was suggested to remove them.

Reply:
For the `if` I'd leave it without, but for the `for` I'd put them in. Chaining the single expressions feels wrong... and I would be surprised if someone actively made the outer braces be removed (and I'd argue against it myself).